File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack=  []
    for ch in s :
        if ch == '(' :
            stack.append(ch)
        elif ch == ')' :
            if len(stack) == 0 :
                return False
            else :
                stack.pop()
                answer =True
    
    if len(stack) == 0 and answer :
        return True
    
    else :
        return False
        
                

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;

       Stack <Character> stack = new Stack<>();
        
        for (int i= 0; i <s.length(); i ++) {
            char  target = s.charAt(i);
            
            if (target == '(') {
                stack.push (target);
            }
            else {
                if (stack.isEmpty()){
                    answer = false;
                }else {
                    if (stack.peek () == '(') {
                        stack.pop();
                    }  else {
                        answer = false;
                    }
                }
            }
        }
        
        if (! stack.isEmpty()) {
            answer = false;
        }
        
        
        return answer;
        
        
        
        

    }
}